sound/soc/sof/intel/hda-dsp.c

```diff
@@ -355,6 +369,13 @@ static int hda_resume(struct snd_sof_dev *sdev)
 	snd_sof_pci_update_bits(sdev, PCI_TCSEL, 0x07, 0);
 
 #if IS_ENABLED(CONFIG_SND_SOC_SOF_HDA)
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_HDA_AUDIO_CODEC)
+	if (runtime_resume) {
+		/* read STATESTS before controller reset */
+		status = snd_hdac_chip_readw(bus, STATESTS);
```

Comment:
here you could also read the status with SOF_HDA directly, even if you don't need it. too many I#if IS_ENABLED